### ISSUE TITLE
Variable not allowed in type expression - type(mulitprocessing.Manager().Queue())

### ISSUE THREAD

**Issue by avoyeux** [microsoft/pyright#8129]
## Environment data

-   Language Server version: v2024.6.1
-   OS and version: Ubuntu 23.10
-   Python version (& distribution if applicable, e.g. Anaconda): python 3.11.6

## Code Snippet

```python
from __future__ import annotations

from typeguard import typechecked
from multiprocessing import Manager

# Determining the type for a multiprocessing.Manager.Queue object
manager = Manager()
queue = manager.Queue()
QUEUE = type(queue)
manager.shutdown()


@typechecked
def main(queue: QUEUE | None = None) -> None:
    """Just a show case empty function.

    Args:
        queue (QUEUE | None): a multiprocessing.Manager.Queue() object. Defaults to None.
    """        

    queue.get()

```

[requirements.txt](https://github.com/user-attachments/files/15803093/requirements.txt)

## Expected behavior

The type annotation is right, at least it passes the @typeguard.typechecked running-time type annotation checks.
Furthermore, in older versions of Pylance (personnaly, I am using v2023.12.1 on a server with the exact same code) there are no Pylance warnings and everything runs smoothly.

## Actual behavior

There is a warning for the type annotation being:
[{
	"resource": "/home/avoyeux/Documents/Cosmic_removal/pylance_issue.py",
	"owner": "_generated_diagnostic_collection_name_#2",
	"code": {
		"value": "reportInvalidTypeForm",
		"target": {
			"$mid": 1,
			"path": "/microsoft/pyright/blob/main/docs/configuration.md",
			"scheme": "https",
			"authority": "github.com",
			"fragment": "reportInvalidTypeForm"
		}
	},
	"severity": 4,
	"message": "Variable not allowed in type expression",
	"source": "Pylance",
	"startLineNumber": 14,
	"startColumn": 17,
	"endLineNumber": 14,
	"endColumn": 22
}]

i.e. it is saying "QUEUE" is not allowed in type expression.

## Logs

Not sure if a log is actually applicable here as I have already copied the issue. Will add it if needed (it's my first ever issue commit so kind of lost).

**Comment by erictraut** [comment]
Pyright is correct to generate an error in this case. As the error indicates, type expressions (i.e. annotations) are not allowed to include variables. While the use of dynamic expressions may work at runtime, it will not work for static type analysis, and the [Python typing spec](https://typing.readthedocs.io/en/latest/spec/annotations.html) is clear that a static type checker should generate an error in this case.

The correct static type annotation in this case is:

```python
from queue import Queue

from typeguard import typechecked

@typechecked
def main(queue: Queue | None = None) -> None:
    queue.get()
```
